REPOSITORY_OVERVIEW.md

```diff
@@ -0,0 +1,1730 @@
+- Huanyang protocol variants
+- Modbus RTU communication
+
+### 5. Kinematics (`FluidNC/src/Kinematics/`)
+
+**Purpose**: Transform between cartesian and motor space
+
+**Supported Kinematics**:
+- `Cartesian` - Standard XYZ motion (most common)
+- `CoreXY` - CoreXY motion system
+- `WallPlotter` - Cable-driven wall plotters
+- `ParallelDelta` - Delta robot kinematics
+- `Midtbot` - SCARA-like robot
+
+### 6. Motion Control
+
+**Key Files**:
+- `GCode.cpp/h` - G-code parser and interpreter
+- `MotionControl.cpp/h` - Motion planning and execution
+- `Planner.cpp/h` - Motion planner (trajectory planning)
+- `Stepper.cpp/h` - Stepper pulse generation
+- `Stepping.cpp/h` - Stepping engine abstraction
+- `Limits.cpp/h` - Limit switch handling
+- `Probe.cpp/h` - Probing operations
+- `Jog.cpp/h` - Jogging mode
+
+### 7. Web Interface (`FluidNC/src/WebUI/`)
+
+**Purpose**: Web-based control interface
+
+**Components**:
+- `WSChannel.cpp/h` - WebSocket communication
+- `TelnetServer.cpp/h` - Telnet access
+- `WebClient.cpp/h` - HTTP client
+- `WebCommands.cpp` - Web command handlers
+- `NotificationsService.cpp/h` - Push notifications
+- `OTA.cpp` - Over-the-air firmware updates
+- `Authentication.cpp/h` - User authentication
+- `Mdns.cpp/h` - mDNS/Bonjour support
+
+### 8. Pin Management (`FluidNC/src/Pins/`)
+
+**Purpose**: Hardware pin abstraction and mapping
+
+**Features**:
+- GPIO pin mapping
+- I2S output pins
+- Virtual pins
+- Pin attributes (pullup, pulldown, invert)
+- Pin parser for configuration strings
+
+---
+
+## Build System
+
+### PlatformIO Configuration
+
+**Main Configuration**: `platformio.ini`
+
+**Build Environments**:
+1. **wifi** (default) - ESP32 with WiFi support
+2. **bt** - ESP32 with Bluetooth support
+3. **noradio** - ESP32 without wireless
+4. **wifi_s3** - ESP32-S3 with WiFi
+5. **bt_s3** - ESP32-S3 with Bluetooth
+6. **noradio_s3** - ESP32-S3 without wireless
+7. **debug** - Debug build with symbols
+8. **tests** - Native unit tests
+
+### Key Build Settings
+
+- **Platform**: espressif32 (Arduino framework)
+- **C++ Standard**: C++17 (`-std=gnu++17`)
+- **Upload Speed**: 921600 baud
+- **Monitor Speed**: 115200 baud
+- **Partition Scheme**: Custom (`min_littlefs.csv`)
+- **Filesystem**: LittleFS
+
+### Dependencies
+
+**External Libraries**:
+- `TMCStepper` (>=0.7.0, <1.0.0) - Trinamic driver support
+- `ESP8266 and ESP32 OLED driver for SSD1306` - Display support
+- `arduinoWebSockets` - WebSocket communication (custom fork)
+- `WiFi` library (custom fork)
+- ESP32 framework libraries (ArduinoOTA, DNSServer, ESPmDNS, WebServer)
+
+### Building the Project
+
+```bash
+# Build default (wifi) environment
+pio run
+
+# Build specific environment
+pio run -e wifi_s3
+
+# Upload firmware
+pio run -t upload
+
+# Upload filesystem
+pio run -t uploadfs
+
+# Run tests
+pio test
+```
+
+### Version Management
+
+The build system uses `git-version.py` to automatically inject git version information into the firmware.
+
```

Comment:
The most important aspects of version management are embodied by "build-release.py" and the manual.yml github actions workflow that creates and publishes release builds.